src/com/google/appengine/demos/shardedcounter/java/v2/ShardedCounter.java

```diff
@@ -234,6 +242,13 @@ private void incrementPropertyTx(final Key key, final String prop,
             LOG.log(Level.WARNING,
                     "You may need more shards. Consider adding more shards.");
             LOG.log(Level.WARNING, e.toString(), e);
+            
+            if(getShardCount() <= ATMOST_SHARDS){
+                LOG.log(Level.INFO,
+                       "Doubling Shade, for this process");
+                addShards(getShardCount()); //double the shade by adding the number of shade again
```

Comment:
yea, will update the naming covention, so it will be MAX_SHARDS so even when it is automatically generating the shard it shouldn't pass the MAX, else for every concurrent exception, so it don't create